sdk/Sdk.Generators/FunctionMetadataProviderGenerator/KnownFunctionMetadataTypes.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
+{
+    internal readonly struct KnownFunctionMetadataTypes
+    {
+        internal readonly INamedTypeSymbol? BindingAttribute;
```

Comment:
same suggestion here: would be slightly preferable to have these all be lazily initialized properties. Also, I recommend exposing properties and not fields - even when they are internal.